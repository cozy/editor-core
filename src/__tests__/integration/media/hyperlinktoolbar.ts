import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { editable, getDocFromElement, fullpage } from '../_helpers';
import {
  mountEditor,
  goToEditorTestingWDExample,
} from '../../__helpers/testing-example-helpers';

const linkUrlSelector = '[data-testid="link-url"]';
const linkLabelSelector = '[data-testid="link-label"]';

async function setupEditor(client: any): Promise<Page> {
  const page = await goToEditorTestingWDExample(client);
  await mountEditor(page, {
    appearance: fullpage.appearance,
  });

  await page.click(editable);
  return page;
}

BrowserTestCase(
  'Inserts a link via hyperlinktoolbar',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const page = await setupEditor(client);

    await page.keyboard.type('k', ['Mod']);

    await page.waitForVisible(linkUrlSelector);
    await page.click(linkUrlSelector);
    await page.keys('www.atlassian.com'.split(''));

    await page.click(linkLabelSelector);
    await page.keys('Hello world!'.split(''));

    await page.keys(['Enter']);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  "Doesn't insert a link via hyperlinktoolbar when switching between fields",
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const page = await setupEditor(client);

    await page.keyboard.type('k', ['Mod']);

    await page.waitForVisible(linkUrlSelector);
    await page.click(linkUrlSelector);
    await page.keys('www.atlassian.com'.split(''));

    await page.click(linkLabelSelector);
    await page.click(linkUrlSelector);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  "Doesn't insert a link via hyperlinktoolbar when clicking out of toolbar",
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const page = await setupEditor(client);

    await page.keyboard.type('k', ['Mod']);

    await page.waitForVisible(linkUrlSelector);
    await page.click(linkUrlSelector);
    await page.keys('www.atlassian.com'.split(''));

    await page.click(editable);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'Inserts a link when tabbing through hyperlinktoolbar',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const page = await setupEditor(client);

    await page.keyboard.type('k', ['Mod']);

    await page.waitForVisible(linkUrlSelector);
    await page.keys('www.atlassian.com'.split(''));
    await page.keys(['Tab']);
    await page.keys('Hello world!'.split(''));
    await page.keys(['Tab']);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
