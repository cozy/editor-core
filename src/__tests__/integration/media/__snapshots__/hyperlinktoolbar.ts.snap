// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doesn't insert a link via hyperlinktoolbar when clicking out of toolbar 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`Doesn't insert a link via hyperlinktoolbar when switching between fields 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`Inserts a link via hyperlinktoolbar 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "http://www.atlassian.com",
              },
              "type": "link",
            },
          ],
          "text": "Hello world!",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`Inserts a link when tabbing through hyperlinktoolbar 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "http://www.atlassian.com",
              },
              "type": "link",
            },
          ],
          "text": "Hello world!",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
